src/autoscaler/cf/app.go

```diff
@@ -39,8 +41,28 @@ func (c *cfClient) GetApp(appId string) (*models.AppEntity, error) {
 	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
-		err = fmt.Errorf("failed getting application summary: %s [%d] %s", url, resp.StatusCode, resp.Status)
-		c.logger.Error("get-app-instances-response", err)
+		respBody, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			c.logger.Error("failed-to-read-response-body-while-getting-app-summary", err, lager.Data{"appid": appId})
+			return nil, err
+		}
+		var bodydata map[string]interface{}
+		err = json.Unmarshal([]byte(respBody), &bodydata)
+		if err != nil {
+			c.logger.Error("failed-to-unmarshal-response-body-while-getting-app-summary", err, lager.Data{"appid": appId})
```

Comment:
1. Besides resp.StatusCode != http.StatusOK , could we check whether resp.StatusCode == 404 first before parsing the response body ?

2.  When CC API is available,  I think the response  error should be written in JSON format.  But if CC API endpoint is not accessible here ,  the response body maybe empty or plain text . In this case, most likely the unmarshal error message talks about the json parsing error , instead of the original http response.  How about adding respBody into the c.logger.Error ?